scripts/Diffraction/isis_powder/gem.py

```diff
@@ -33,11 +33,12 @@ def focus(self, **kwargs):
         self._inst_settings.update_attributes(kwargs=kwargs)
         self._switch_texture_mode_specific_inst_settings(kwargs.get("texture_mode"))
 
-        return self._focus(
+        focused_runs = self._focus(
             run_number_string=self._inst_settings.run_number,
             do_van_normalisation=self._inst_settings.do_van_norm,
             do_absorb_corrections=self._inst_settings.do_absorb_corrections,
         )
+        return self._output_focused_runs(focused_runs, self._inst_settings.run_number)
```

Comment:
I see that you've moved the file output out of the `_focus_one_ws` function in focus.py because you need to delay this on OSIRIS until after you've done the dRange merge. That general aim is fine but I think that you can achieve this without editing the gem\polaris\pearl\hrpd.py units. 

I think you can add this call to `self._output_focused_runs` into the _focus function in the abstract_inst.py unit + create an overridden version of _focus in the osiris.py unit.

OSIRIS is the special case (odd one out) here so it feels like it should be the osiris.py unit that overrides the common code to stop the output happening early. I think my suggestion makes this a bit more clear.

Reply:
This is fixed now